### DISCUSSION TITLE
How do I customize the default input field and Done button

### DISCUSSION THREAD

**Discussion by ShivamJoker** [botui/botui#180] [Q&A]
Thanks for this awesome library!

I am unable to find docs which can help me customize the input component.

![image](https://user-images.githubusercontent.com/23727670/216642808-cbcacb73-e921-44bf-ae13-236b8f0f75c8.png)

I am using the React library and I want to add an icon instead of text, can anyone help me please?

**Comment by moinism** [comment]
Hey @ShivamJoker,

I'm glad BotUI is helpful for you.

Regarding styling, I must admit it's not the most straightforward thing to do now with default components. So I'm considering making it a lot more intuitive.

To achieve what you've listed, the best way would be to create a custom `action` for `input`. This means your custom component will be shown as the input, and you can style it however you want. Here're the [docs on how to do that](https://docs.botui.org/react/custom/#add-a-custom-action).

I hope it helps.
